hawkbit-rest/hawkbit-ddi-resource/src/test/java/org/eclipse/hawkbit/ddi/rest/resource/DdiInstalledBaseTest.java

```diff
@@ -0,0 +1,411 @@
+import org.eclipse.hawkbit.repository.event.remote.entity.TargetUpdatedEvent;
+import org.eclipse.hawkbit.repository.model.Action;
+import org.eclipse.hawkbit.repository.model.ActionStatus;
+import org.eclipse.hawkbit.repository.model.Artifact;
+import org.eclipse.hawkbit.repository.model.DistributionSet;
+import org.eclipse.hawkbit.repository.model.Target;
+import org.eclipse.hawkbit.repository.test.matcher.Expect;
+import org.eclipse.hawkbit.repository.test.matcher.ExpectEvents;
+import org.eclipse.hawkbit.rest.util.JsonBuilder;
+import org.eclipse.hawkbit.rest.util.MockMvcResultPrinter;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.hateoas.MediaTypes;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.ResultActions;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import io.qameta.allure.Description;
+import io.qameta.allure.Feature;
+import io.qameta.allure.Story;
+
+/**
+ * Test installed base from the controller.
+ */
+@Feature("Component Tests - Direct Device Integration API")
+@Story("Installed Base Resource")
+public class DdiInstalledBaseTest extends AbstractDDiApiIntegrationTest {
+
+    private static final int ARTIFACT_SIZE = 5 * 1024;
+    private static final String CONTROLLER_ID = "4715";
+
+    @Test
+    @Description("Ensure that the installed base resource is available as CBOR")
+    public void installedBaseResourceCbor() throws Exception {
+        final Target target = testdataFactory.createTarget();
+        final DistributionSet ds = testdataFactory.createDistributionSet("");
+        final Long actionId = getFirstAssignedActionId(assignDistributionSet(ds, target));
+        postDeploymentFeedback(target.getControllerId(), actionId,
+                JsonBuilder.deploymentActionFeedback(actionId.toString(), "closed"), status().isOk());
+
+        // get installed base
+        performGet(INSTALLED_BASE, MediaType.parseMediaType(DdiRestConstants.MEDIA_TYPE_CBOR), status().isOk(),
+                tenantAware.getCurrentTenant(), target.getControllerId(), actionId.toString());
+
+        final Long softwareModuleId = ds.getModules().stream().findAny().get().getId();
+        testdataFactory.createArtifacts(softwareModuleId);
+        // get artifacts
+        performGet(SOFTWARE_MODULE_ARTIFACTS, MediaType.parseMediaType(DdiRestConstants.MEDIA_TYPE_CBOR),
+                status().isOk(), tenantAware.getCurrentTenant(), target.getControllerId(),
+                String.valueOf(softwareModuleId));
+    }
+
+    @Test
+    @Description("Test several deployments to a controller. Checks that action is represented as installedBase after installation.")
+    public void deploymentSeveralActionsInInstalledBase() throws Exception {
```

Comment:
It is possible to also look up previous installedBase action IDs, but your tests only check for the latest (as far as I can see). I would suggest to either extend this test or add another one were you check that looking up older installedBase actionIds works.

Reply:
Done. Added another call to the older action to the same test